daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala

```diff
@@ -133,6 +132,15 @@ class EngineTest
         withKeyContractInst,
     )
 
+  val defaultKey = Map(
```

Comment:
Was this just broken before? I’m a bit confused why we now need to add a global key that we did not need before.

Reply:
Before `1.14`, `by-key` node where replayed `by-id`, so we did not need to mock the `ResultNeedKey` lookup during reinterpretation. 

Now we have the `by-key` flag we replay them `by-key`, so we need to mock properly the  `ResultNeedKey` lookup.

Reply:
oh I see, so the state was broken before but because we replayed by-id speedy did not emit a SResultNeedKey. Thanks for the explanation